dts/bindings/arm/nordic,nrf-uicr.yaml

```diff
@@ -7,3 +7,12 @@ include: base.yaml
 properties:
   reg:
     required: true
+
+  nfct-pins-as-gpios:
```

Comment:
Wouldn't it be better to add this property to the `nordic,nrf-pinctrl` binding instead? UICR is just a storage for this setting (in the existing SoCs, and hypothetically this may change in some future ones) but the setting itself defines function of certain pins, so its related with what the `pinctrl` node is supposed to do.

Reply:
I think this approach sticks close to the hardware, on nrf52/53 it's a UICR property, and so I think `uicr` node is what makes more sense. If newer SoCs have this managed by NFCT, then the property can be part of `nfct` compatible. If that's true, new SoCs will also have a new compatible for UICR, e.g. `nordic,nrf-uicr-v2` because it will be a different piece of IP.

Reply:
@hubertmis @57300 @nordicjm @doki-nordic @KAGA164 @dchat-nordic Could you please give your opinion here?

Reply:
I'm of the mind that is belongs in UICR since that is what sets and controls it. If this was moved to pinctrl then it was moved in a future chip, the pinctrl would need to be uprev'd instead of UICR but pinctrl might not have changed at all

Reply:
I vote for having it under uicr for now. Seems convenient, and moving it out of Kconfig is already an improvement.

I can see pinctrl being the right choice though, no matter which IP contains the setting. But first, we could consider whether it makes sense to move the relevant `SystemInit()` code into the pinctrl driver. For example, could there be a use case for reconfiguring `NFCPINS` between runs, instead of hardcoding it?

Reply:
@nordicjm and @57300 would you please approve the PR in this case?